dogma.go

```diff
@@ -390,13 +390,22 @@ func (c *Client) Push(ctx context.Context, projectName, repoName, baseRevision s
 // WatchFile awaits and returns the query result of the specified file since the specified last known revision.
 func (c *Client) WatchFile(ctx context.Context, projectName, repoName, lastKnownRevision string,
 	query *Query, timeout time.Duration) <-chan *WatchResult {
-	return c.watch.watchFile(ctx, projectName, repoName, lastKnownRevision, query, timeout)
+	watchResult := make(chan *WatchResult)
+	go func() {
+		watchResult <- c.watch.watchFile(ctx, projectName, repoName, lastKnownRevision, query, timeout)
```

Comment:
We got the result and the routine exit. so file would be no watch anymore?

I think user expect to keep an eye on `watchResult`, until `watchResult` is closed.

Reply:
Well, this `WatchFile` is just for one time only.
We provide a `FileWatcher` to watch endlessly for users.

Reply:
The `watchFile` which watches one time only and `FileWatcher` which keeps watching came from the Java version that works exactly in the same way. But now I thought about it more and we can consolidate the API into one which just returns a channel so that the caller just receives the result from it until it's closed. WDYT @trustin?

Reply:
Yes I think so. Because one time watch is no different that just get-after-sometime.

So I think it's better to do make WatchFile return a Watcher

Reply:
And because we pass in a `ctx`, so it's easy to stop the watcher by letting user cancel `ctx` for us. We just don't care about how to stop it.

Reply:
Yeah right~ It' already implemented in the API :)

Reply:
Yeah,  I saw it at `Watcher` 👍 
That's why I think that we just return a `Watcher` is enough and easy to fix :)

Reply:
@linxGnu I thought about your comments and had a chat with @trustin. It turned out to be that we don't need separate APIs and we can only provide an integrated API for one time or whole time watch. Will fix soon. Thank you~!